src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -0,0 +1,221 @@
+    maxDiff = None
+
+    @classmethod
+    def setUpTestData(cls):
+        ObjectsAPIConfigFactory.create(
+            objects_service__api_root="https://objecten.nl/api/v2/",
+            objecttype="https://objecttypen.nl/api/v1/objecttypes/1",
+            objecttype_version=1,
+            productaanvraag_type="terugbelnotitie",
+        )
+
+    def test_submission_with_objects_api_backend_override_defaults(self, m):
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "voornaam",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_voornamen,
+                    },
+                },
+                {
+                    "key": "achternaam",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_geslachtsnaam,
+                    },
+                },
+                {
+                    "key": "tussenvoegsel",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_tussenvoegsel,
+                    },
+                },
+                {
+                    "key": "geboortedatum",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_geboortedatum,
+                    },
+                },
+            ],
+            submitted_data={
+                "voornaam": "Foo",
+                "achternaam": "Bar",
+                "tussenvoegsel": "de",
+                "geboortedatum": "2000-12-31",
+            },
+            bsn="111222333",
+        )
+
+        mock_service_oas_get(m, "https://objecten.nl/api/v2/", "objecten")
+
+        objects_form_options = dict(
+            objecttype="https://objecttypen.nl/api/v1/objecttypes/2",
+            objecttype_version=2,
+            productaanvraag_type="testproduct",
+        )
+
+        expected_result = {
+            "url": "https://objecten.nl/api/v1/objects/1",
+            "uuid": "095be615-a8ad-4c33-8e9c-c7612fbf6c9f",
+            "type": objects_form_options["objecttype"],
+            "record": {
+                "index": 0,
+                "typeVersion": objects_form_options["objecttype_version"],
+                "data": {
+                    "data": submission.get_merged_data(),
+                    "type": objects_form_options["productaanvraag_type"],
+                    "submission_id": str(submission.uuid),
+                },
+                "geometry": {"type": "Point", "coordinates": [0, 0]},
+                "startAt": date.today().isoformat(),
+                "endAt": date.today().isoformat(),
+                "registrationAt": date.today().isoformat(),
+                "correctionFor": 0,
+                "correctedBy": "",
+            },
+        }
+
+        m.post(
+            # FIXME zgw-consumers seems to duplicate /api/v2
```

Comment:
so that effectively makes this PR broken, right?

Reply:
For Objects API v2 yes (though v2 is not released yet). The spec of objects v2 has `/api/v2/` in the paths (which won't work with zgw-consumers) like this:

```yaml
/api/v2/objects:
    get:
      operationId: object_list
      description: Retrieve a list of OBJECTs and their actual RECORD. The actual
        record is defined as if the query parameter `date=<today>` was given.
```

Is this an issue in the objects v2 API spec or should this be addressed in zgw-consumers?

I guess for this PR I could switch to objects v1?

Reply:
I asked Anna and she said Silvia made an issue about this in objects API (https://github.com/maykinmedia/objects-api/issues/226). This API spec change was introduced in 1.1.1: https://github.com/maykinmedia/objects-api/blob/1.1.1/src/openapi.yaml#L71, so the Objects API registration should still work with Objects API <= 1.1.0. For the tests, I changed the Objects OAS to 1.1.0

Is there anything I can do/should do on the openforms/zgw-consumers side about this?

Reply:
The changes to remove `/api/v[0-9]` are merged to the Objects API master branch (and are available through 'latest' tag I believe), so it should not block you anymore

Reply:
I'm not sure what this means, but let's target the latest version of Objects API (`latest` docker image tag).

You can pin the API spec in Open Forms to the commit that is now `master` and please at a comment referring back to this discussion.

Once v2.0 is out, we'll update the tags/images etc.

Reply:
Mentioned in https://github.com/open-formulieren/open-forms/pull/501/commits/c297718ced70d98a520cb4b64afb9082f7009716